pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +436,40 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
+	defer trace.End(trace.Begin(""))
+
+	var vms []mo.VirtualMachine
+
+	// Retrieve config.hardware property of all machines under this view
+	err := m.view.Retrieve(op, []string{"VirtualMachine"}, []string{"config.hardware", "runtime.powerState"}, &vms)
+	if err != nil {
+		return nil, err
+	}
+
+	// iterate over them to see whether they have the disk we want
+	for i := range vms {
+		name := vms[i].Name
+		state := vms[i].Runtime.PowerState
+		if state != types.VirtualMachinePowerStatePoweredOn && state != types.VirtualMachinePowerStateSuspended {
```

Comment:
it is filtered out but I'm going to change this filtering logic like George suggested https://github.com/vmware/vic/pull/5627#discussion_r125130965